src/ansys/fluent/core/solver/flobject.py

```diff
@@ -1067,9 +1067,10 @@ def __setattr__(self, name: str, value):
         try:
             return attr.set_state(value)
         except Exception as ex:
-            allowed = attr.allowed_values()
-            if allowed and value not in allowed:
-                raise allowed_values_error(name, value, allowed) from ex
+            if hasattr(attr, "allowed_values"):
+                allowed = attr.allowed_values()
+                if allowed and value not in allowed:
+                    raise allowed_values_error(name, value, allowed) from ex
             else:
                 raise ex
```

Comment:
Wouldn't a plain `raise` be sufficient here?

Reply:
@h-krishnan Yes, but we think that it looks safe.

Reply:
Done. Thank you.